scipy/stats/_stats_py.py

```diff
@@ -3135,9 +3147,12 @@ def gzscore(a, *, axis=0, ddof=0, nan_policy='propagate'):
     >>> plt.show()
 
     """
-    a = np.asanyarray(a)
-    log = ma.log if isinstance(a, ma.MaskedArray) else np.log
-
+    if isinstance(a, ma.MaskedArray):
+        log = ma.log
+    else:
+        xp = array_namespace(a)
+        a = _convert_common_float(a, xp=xp)
```

Comment:
Is this needed because `zmap` calls `_convert_common_float`

Reply:
Yes, for `log`.

Reply:
Sorry I meant `a = _convert_common_float(a, xp=xp)`, is this not duplicated with line 3228?

Reply:
Oh wait I see what you mean